### PR TITLE
Closes #84 xportr deep dive vignette

### LINKED ISSUE #84: Detailed vignette of how to use xportr
### Overview 
* In-depth discussion of submission process
* Break down the specification file and how they fit into `xportr` functions
  * Possible opportunity to use Mermiad or some other diagramming software
* Discuss Assumptions of `xportr` functions
  * Example: assumptions around length in spec and in function
  * Discuss use of `xportr` options to address different specification file styles
  * Discuss use of `xportr_set()` to reduce repetitive tasks (#44)
  * Discuss `metacore` versus dataframe spec option

### ADaMs and Specs
* Discuss Pilot 3 Submission as source of data
* Where is data located for users to experiment with.
* Read in data and briefly highlight features.
* Read in specs and briefly highlight features

### Application of xportr functions to one ADaM dataset (TBD)
* Spec should be altered at times to help highlight messaging/warnings built it to `xportr`

### Application of xportr functions to all ADaM datasets (TBD)
* Section will be repeat of above so should just be hand-wavey 
* No highlighting of message/warnings
* Goal is to show bulk application of functions to multiple datasets

### Validation Software
* Showcase readouts from community P21 Tool

## Outstanding Questions
* Where does the specs and datasets live? **Decision**: Fine in `example_data_sepc`
* It would be nice to get some outside reviews done before we submit to CRAN.

## Definition of Done
A detailed and user-reviewed vignette available within the package and on the site

### PR DESCRIPTION
**Pull request by bms63**
### Thank you for your Pull Request! 

We have developed a Pull Request template to aid you and our reviewers.  Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the xportr codebase remains robust and consistent.  

### The scope of `{xportr}`

`{xportr}`'s scope is to enable R users to write out submission compliant `xpt` files that can be delivered to a Health Authority or to downstream validation software programs.  We see labels, lengths, types, ordering and formats from a dataset specification object (SDTM and ADaM) as being our primary focus. We also see messaging and warnings to users around applying information from the specification file as a primary focus. Please make sure your Pull Request meets this **scope of {xportr}**. If your Pull Request moves beyond this scope, please get in touch with the `{xportr}` team on [slack](https://pharmaverse.slack.com/archives/C030EB2M4GM) or create an issue to discuss.

Please check off each task box as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

### Changes Description

Vignette can be found here: https://bms63.github.io/xportr/articles/deepdive.html

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description.  Can be removed or left blank if changes are minor/self-explanatory.
- [x] Check that your Pull Request is targeting the `devel` branch, Pull Requests to `main` should use the [Release Pull Request Template](https://github.com/atorus-research/xportr/tree/94_pr_template/.github/PULL_REQUEST_TEMPLATE)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/).  Use `styler` package and functions to style files accordingly.
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Address any updates needed for vignettes and/or templates
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] Fix merge conflicts 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: vignettes/deepdive.Rmd [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+    columnDefs = list(list(visible = FALSE, targets = columns2hide )))
+  )
+```
+
+### `xportr_type()`
+
+We are going to explore the type column in the metadata object. A submission to a Health Authority should only have character and numeric types in the data. In the  `ADSL` data we will have several columns that are in the Data type: `TRTSDT`, `TRTEDT`, `DISONSDT`, `VISIT1DT` and `RFENDT` and we will change one variable type to a factor variable. 
+
+```{r}
+adsl_loc <- here::here("example_data_specs", "adsl.xpt")
```
Comment (averissimo) [inline vignettes/deepdive.Rmd:198]: I think you missed the `adsl.xpt` file when adding to git, probably due the entry on the `.gitignore` file

Comment (bms63) [reply vignettes/deepdive.Rmd:198]: whoops yeah that shouldn't be there!  it is actually from writing out the file

**Comment by bms63** [comment]
Hey all - hopefully 90% there!!

Here is https://bms63.github.io/xportr/articles/deepdive.html site with vignette to ease in the review.  @adchan11 I know you are busy, but hopign you can give some pointers as Roche is consumer of this package :)

File: vignettes/xportr.Rmd [hunk 2]
```diff
@@ -45,7 +56,7 @@ The `ADSL` has the following features:
 * Data types other than character and numeric
 * Missing labels on variables
 * Missing label for data set
-* Order of varibles not following specification file
+* Order of variables not following specification file
```
Comment (bms63) [inline vignettes/xportr.Rmd:59]:
```suggestion
* Order of variables not following specification files
```

**Review by EeethB — COMMENTED** [review]
🏅 What a wealth of information! I'm sure this will make xportr adoption much easier. Here's an overview of the changes I made. I feel like the review process here makes the suggestions look massive because it repeats each paragraph twice, but it's really just some tweaks :grimacing: :grimacing: :grimacing:

* 🔠 Fixed a couple typos
* :hash: I'm very opinionated about commas, so I added or deleted a whole bunch of them - of course feel free to ignore any suggested changes! :)
* ⏩ Aligned wording on 'the xportr_*() function such-and-such...' vs just 'xportr_*() such-and-such...'. There was a blend of both before - Let me know if you prefer the first, and I can align on that
* 👩‍💻 Standardized to double quotes in code
* 📦 Loaded all packages used explicitly and got rid of double-colon notation - I think for reproducibility it would also be best to pull `library()` calls out of that first chunk that's not included. I got some errors copying over code and running it because I didn't have rlang or haven loaded initially.
* 💪 Strengthened some assertions that (I think) we're committed to keeping
* 🔀 Made some other rewording changes - again feel free to ignore if you prefer!

🙏 This was really helpful to read through, it gave me a better sense of how everything fits together!

File: vignettes/deepdive.Rmd [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = " "
+)
+
+options(cli.num_colors = 1)
+
+library(DT)
+library(rlang)
+library(haven)
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:23]: i have kept DT hidden as users don't need that package.  But I put rlang, dplyr and xportr all in the same code chunk

**Comment by bms63** [comment]
Hi all,

Ethan has given me great feedback and I have implemented it.  You can review vignette on the site https://bms63.github.io/xportr/articles/deepdive.html

File: vignettes/deepdive.Rmd [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+The next two examples showcase the `strict_checks = TRUE` option in `xportr_write()` where we will look at formats and labels.
+
+```{r, echo = TRUE, error = TRUE}
+adsl %>%
+  xportr_write(path = "adsl.xpt", label = "Subject-Level Analysis Dataset", strict_checks = TRUE)
+```
+
+
+As there at several `---DT` type variables, `xportr_write()` detects the lack of formats being applied. To correct this remember you can use `xportr_type()` and `xportr_format()` to apply formats to your xpt dataset.
+
```
Comment (cpiraux) [inline vignettes/deepdive.Rmd:383]: The message for format is due to DATE9., it is not recognized as a SAS format for the moment.

File: vignettes/deepdive.Rmd [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+* `options()` 
+* `xportr_metadata()` 
+
+**NOTE:** As long as you have a well-defined _metadata object_ you do NOT need to use `options()` or `xportr_metadata()`, but we find these handy to use and think they deserve a quick mention!
+
+## You got `options()`
+
+`{xportr}` is built with certain assumptions around specification column names and information in those columns. We have found that each company specification file can differ slightly from our assumptions. For example, one company might call a column `Variables`, another `Variable` and another `variables`.  Rather than trying to regex ourselves out of this situation, we have introduced `options()`.  `options()` allows users to control those assumptions inside `{xportr}` functions based on their needs.
+
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:110]: I was thinking that we need a function to display these as the `zzz.R` is a bit raw to look at - #130

File: vignettes/deepdive.Rmd [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+  var_spec,
+  rownames = FALSE,
+  filter = "top",
+  options = list(
+    dom = "Bfrtip",
+    columnDefs = list(list(visible = FALSE, targets = columns2hide))
+  )
+)
+```
+
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:198]: @averissimo any DT tips you can drop us?

[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:198]: probably use these throughout the packagedown site

File: vignettes/deepdive.Rmd [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+Just like we did for the other functions, setting `verbose = "stop"` immediately stops R from processing the order. If variables or metadata are missing from either, the re-ordering will not process until corrective action is performed.
+
+## `xportr_format()`
+
+Formats play an important role in the SAS language and have a column in specification files. Being able to easily apply formats into your `xpt` file will allow downstream users of SAS to quickly format the data appropriately when reading into a SAS-based system.  `xportr_format()` can take these formats and apply them. Please reference `xportr_length()` or `xportr_label()` to note the missing `attr()` for formats in our `ADSL` dataset.
+
+This example is slightly different from previous examples. You will need to use `xportr_type()` to coerce R Date variables and others types to character or numeric. Only then can you use `xportr_format()` to apply the format column to the dataset.
+
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:340]: I actually think if the variable is a Date then `xportr_format()` won't apply the format

File: vignettes/deepdive.Rmd [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+```
+
+Success! We have taken the metadata formats and applied them to the dataset. Please inspect variables like `TRTSDT` or `DISONSDT` to see the `DATE9.` format being applied.
+
+```{r, max.height='300px', attr.output='.numberLines', echo = FALSE}
+str(adsl_fmt)
+```
+
+At the time of `{xportr} v0.3.0` we have not impleneted any warnings or error messaging for this function. However, `xportr_write()` through `xpt_validate()` will check that formats have been applied for Date-Time variables.
+
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:354]: ah okay I will make a change to this and we can put this in as an issue

File: vignettes/deepdive.Rmd (hunk 5 above)
[the remark this answers is not in the document]
Comment (cpiraux) [reply vignettes/deepdive.Rmd:110]: That sounds good to me

File: vignettes/deepdive.Rmd [hunk 9]
```diff
@@ -0,0 +1,404 @@
...
+## Using `options()` and `xportr_metadata()` to enhance your experience.
+
+Before we dive into the functions, we want to point out some quality of life utilities to make your `xpt` generation life a little bit easier.
+
+* `options()` 
+* `xportr_metadata()` 
+
+**NOTE:** As long as you have a well-defined _metadata object_ you do NOT need to use `options()` or `xportr_metadata()`, but we find these handy to use and think they deserve a quick mention!
+
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:106]: Yes correct.  I was more saying that as long as your metadata meets the basics assumptions for `xportr` then you don't need to use `options` or `xportr_metadata()` just nice quality of life imporvements.

File: vignettes/deepdive.Rmd (hunk 6 above)
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:197]: I managed to format the DataTable and make it a bit more compact 
image

How does that look?

File: vignettes/deepdive.Rmd (hunk 7 above)
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:339]: Here is example of `xportr_format()` not applying formats to the dataframe as a single call if the variables are of the Date class.  If you use `xportr_format()` with `xportr_type()` then it applies the formats

@atorus-research/xportr-development-team not sure if this was intentionally designed like this, but we should make a note in the documentation that these two are needed.  If the variables is already a character then you don't need to use ``xportr_type()1

```
library(lubridate)

adsl <- tribble(
  ~STUDYID,   ~USUBJID,              ~TRTEDTM,      ~TRTEDT,
  "PILOT01", "01-1130", "2014-08-16 23:59:59", "2014-08-16",
  "PILOT01", "01-1133", "2013-04-28 23:59:59", "2013-04-28",
  "PILOT01", "01-1211", "2013-01-12 23:59:59", "2013-01-12",
  "PILOT01", "09-1081", "2014-04-27 23:59:59", "2014-04-27",
  "PILOT01", "09-1088", "2014-10-09 23:59:59", "2014-10-09"
) %>% 
  mutate(TRTEDTM = ymd_hms(TRTEDTM, tz=Sys.timezone()),
         TRTEDT = as.Date(TRTEDT),
         STUDYID = as.factor(STUDYID))

metadata <- data.frame(
  dataset = c("adsl", "adsl", "adsl", "adsl"),
  variable = c("STUDYID", "USUBJID", "TRTEDTM", "TRTEDT"),
  type = c("character", "character", "numeric", "numeric"),
  format = c(NA, NA, "DATETIME20.", "DATE9.")
)

adsl_notype <- xportr_format(adsl, metadata)

adsl_type <- adsl %>% 
  xportr_type(metadata = metadata) %>% 
  xportr_format(metadata = metadata)

```

image

[the remark this answers is not in the document]
Comment (cpiraux) [reply vignettes/deepdive.Rmd:340]: Thanks @bms63 for the example code.

I also ran `xportr_write(adsl_type, "typefmt.xpt") ` and open it in SAS. As the date class is coerced to dbl, the date is incorrect  in SAS:
image


Date class will not be coerced anymore after the update  #145 to avoid incorrect date in SAS.
I tested with the example code in the branch of the PR #145 and the format is not added in adsl_type. 
I guess, we will need to update `xportr_format()`, right?

File: vignettes/deepdive.Rmd (hunk 6 above)
[the remark this answers is not in the document]
Comment (cpiraux) [reply vignettes/deepdive.Rmd:198]: I have the feeling that values in 'Label' are less readable than previously. I would like to have more space between line in the same row.

I like the way datatable are rendered in admiral vignette [admiraldev/dataset_vignette.R](https://github.com/pharmaverse/admiraldev/blob/main/R/dataset_vignette.R)

File: vignettes/deepdive.Rmd [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+<img src="xpt_validate.png" alt="validate" style="width:800px;"/>
+
+
+# {xportr} in action
+
+In this section, we are going to explore the 5 core `{xportr}` functions using: 
+
+* An ADSL ADaM dataset from the Pilot 3 Submission to the FDA
+* The ADSL ADaM Specification File from the Pilot 3 Submission to the FDA
```
Comment (vedhav) [inline vignettes/deepdive.Rmd:91]:
```suggestion
* `xportr::adsl` - An ADSL ADaM dataset from the Pilot 3 Submission to the FDA
* `xportr::var_spec` - The ADSL ADaM Specification File from the Pilot 3 Submission to the FDA
```
It would be nice to have a link to the data source where it is pointed out. WDYT?

File: vignettes/deepdive.Rmd [hunk 11]
```diff
@@ -0,0 +1,404 @@
...
+  xportr_format() %>%
+  xportr_write("adsl.xpt", label = "Subject-Level Analysis Dataset")
+```
+
+
+## Warnings and Errors 
+
+For the next six sections, we are going to explore the Warnings and Errors messages generated by the `{xportr}` core functions. To better explore these, we will either manipulate the ADaM dataset or specification file to help showcase the ability of the `{xportr}` functions to detect issues.  
+
+**NOTE:** We have made the ADSL, `adsl`, and Specificaion File, `var_spec`, available in this package. Users can find additional datasets and specification files on our [repo](https://github.com/atorus-research/xportr) in the `example_data_specs` folder. We have not included these in our package to help keep the package to a minimum size.
```
Comment (vedhav) [inline vignettes/deepdive.Rmd:168]: I think this **NOTE** is outdated (~We have not included these in our package~). The **NOTE** above (~line 95) has the proper text. Although we can change the `adsl` and `var_spec` to `xportr::adsl` and `xportr::var_spec` respectively as this will link it to the data reference.

File: vignettes/deepdive.Rmd [hunk 12]
```diff
@@ -0,0 +1,404 @@
...
+## Using `options()` and `xportr_metadata()` to enhance your experience.
+
+Before we dive into the functions, we want to point out some quality of life utilities to make your `xpt` generation life a little bit easier.
+
+* `options()` 
+* `xportr_metadata()` 
+
+**NOTE:** As long as you have a well-defined _metadata object_ you do NOT need to use `options()` or `xportr_metadata()`, but we find these handy to use and think they deserve a quick mention!
+
+## You got `options()`
```
Comment (vedhav) [inline vignettes/deepdive.Rmd:107]: I think we can also have a small section after the cleverly named _You got options_ like _verbose your choices_ to explain what this does because just like options verbose is something that many functions get affected so it's worth mentioning it before diving into them. WDYT?

**Review by vedhav — COMMENTED** [review]
Love the structure of the deep dive vignette. Especially the fact that now we also ship sample data and spec. It would be much easier to provide reprex with these. I left a few comments PTAL.

File: _pkgdown.yml [hunk 13]
```diff
@@ -18,7 +18,6 @@ navbar:
       href: https://pharmaverse.slack.com/archives/C030EB2M4GM
       aria-label: slack
 
-
 reference:
 - title: The six core xportr functions
 - contents:
```
Comment (vedhav) [inline _pkgdown.yml:23]: The pkgdown yaml config needs to be linted. I use prettier which auto-linted this for me. I can commit this change if you don't mind.

File: vignettes/deepdive.Rmd (hunk 10 above)
Comment (bms63) [reply vignettes/deepdive.Rmd:91]: Yes this is a great idea!!

File: DESCRIPTION [hunk 14]
```diff
@@ -81,3 +81,5 @@ Suggests:
     metacore
 Config/testthat/edition: 3
 VignetteBuilder: knitr
+Depends: 
+    R (>= 2.10)
```
Comment (averissimo) [inline DESCRIPTION:85]: dplyr depends on `≥ 3.5.0`, is there any specific reason for `2.10`?

File: vignettes/deepdive.Rmd [hunk 15]
```diff
@@ -0,0 +1,404 @@
...
+
+Each of the core `{xportr}` functions for applying labels, types, formats, order and lengths provides feedback to users on submission compliance. However, a final check is implemented when `xportr_write()` is called to create the `xpt`. `xportr_write()` calls [`xpt_validate()`](https://github.com/atorus-research/xportr/blob/231e959b84aa0f1e71113c85332de33a827e650a/R/utils-xportr.R#L174), which is a behind the scenes/non-exported function that does a final check for compliance.  At the time of `{xportr} v0.3.0` we are checking the following when a user writes out an `xpt` file.:
+
+<img src="xpt_validate.png" alt="validate" style="width:800px;"/>
```
Comment (averissimo) [inline vignettes/deepdive.Rmd:83]: Force the image to have the same width as the text

```suggestion
<img src="xpt_validate.png" alt="validate" style="width: calc(100% - 2px); padding: 0; margin: 0;"/>
```

**Review by averissimo — COMMENTED** [review]
It seems like the bottom of the vignette is very glitchy. 

Maybe it's only on my system.

Made a bunch of comments with suggestions on how to implement

File: vignettes/deepdive.Rmd (hunk 6 above)
[the remark this answers is not in the document]
Comment (averissimo) [reply vignettes/deepdive.Rmd:198]: I made some suggestion above, this is how it looks

image

File: R/type.R [hunk 16]
```diff
@@ -62,8 +62,11 @@ xportr_type <- function(.df,
 
   ## Pull out correct metadata
   metadata <- metadata %||%
-    attr(.df, "_xportr.df_metadata_") %||%
+    attr(.df, "_xportr.df_metadata_")
```
Comment (averissimo) [inline R/type.R:65]: Why the change here? Was it a matter of being more readable?

I think this implementation is also on other functions

File: vignettes/xpt_validate.png
Comment (averissimo) [inline vignettes/xpt_validate.png:1]: This is a non-standard location for an image file. Maybe move it to `man/figures` and change the `img` tag

I've tried this locally and it seems to work both with the package vignette and the site

```
<img src="../man/figures/xpt_validate.png" alt="validate" style="width: calc(100% - 2px); padding: 0; margin: 0;"/>
```

File: vignettes/deepdive.Rmd [hunk 17]
```diff
@@ -0,0 +1,404 @@
...
+datatable(
+  var_spec,
+  rownames = FALSE,
+  filter = "top",
+  options = list(
+    dom = "Bfrtip",
+    columnDefs = list(list(visible = FALSE, targets = columns2hide))
+  )
+) %>%
+  formatStyle(0, target = "row", color = "black", backgroundColor = "white", fontWeight = "bold", lineHeight = "70%")
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:196]: This is fantastic @cpiraux.  check out amazing work from @averissimo !!.  I will implement this is the Get Started Page as well. TYTY

[the remark this answers is not in the document]
Comment (cpiraux) [reply vignettes/deepdive.Rmd:196]: This looks really nice! Thanks @averissimo @bms63

File: vignettes/deepdive.Rmd (hunk 11 above)
Comment (bms63) [reply vignettes/deepdive.Rmd]: Updated

File: vignettes/deepdive.Rmd (hunk 12 above)
Comment (bms63) [reply vignettes/deepdive.Rmd:107]: haha - how about _"Are we being too verbose?"_

File: vignettes/deepdive.Rmd (hunk 17 above)
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd]: It looks like the pkgdown squeezes the longer labels still but the standalone vignette works as intended.  Going to leave as I think it is overall a big improvement.  

image

File: vignettes/xpt_validate.png
Comment (bms63) [reply vignettes/xpt_validate.png:1]: File has been moved.

File: vignettes/xportr.Rmd [hunk 18]
```diff
@@ -64,114 +69,88 @@ library(dplyr)
...
-<br>
-
-
-
-```{r, echo = TRUE, eval = FALSE}
-str(adsl)
-```
+Next we can apply the lengths from a variable level specification file to the data frame.  `xportr_length()` will identify variables that are missing from your specification file.  The function will also alert you to how many lengths have been applied successfully.  Before we apply the lengths lets verify that no lengths have been applied to the original dataframe.
 
 ```{r, max.height='300px', attr.output='.numberLines', echo = FALSE}
```
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/xportr.Rmd:144]: Unfortunate - I was following this advice https://bookdown.org/yihui/rmarkdown-cookbook/hook-scroll.html

File: vignettes/deepdive.Rmd (hunk 12 above)
Comment (vedhav) [reply vignettes/deepdive.Rmd:107]: :D yessss

File: _pkgdown.yml (hunk 13 above)
Comment (vedhav) [reply _pkgdown.yml]: I pushed the linted file now.
No, Our linter CI only lints R files because we only do `lintr::lint_package()` may be an overkill to check linting for `yaml` files just for one file :)

File: vignettes/xportr.Rmd (hunk 18 above)
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/xportr.Rmd]: Alright this seems to work for the vignette standalone and the pkgodwn site - https://github.com/rstudio/rmarkdown-cookbook/issues/75

[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/xportr.Rmd]: image

File: vignettes/deepdive.Rmd (hunk 12 above)
Comment (bms63) [reply vignettes/deepdive.Rmd:114]: Added a small section

**Comment by bms63** [comment]
Hi all

New website: https://bms63.github.io/xportr/

Will fix failing actions tmr.  Think we are close - still couple of minor issues to iron with formats, but that might be for next release.

File: vignettes/deepdive.Rmd [hunk 19]
```diff
@@ -0,0 +1,438 @@
...
+        sprintf('style="max-height: %s;"', options$max.height)
+      )
+    }
+    hook_output(x, options)
+  })
+})
+```
+
+```{css, echo=FALSE}
+# Used to control DT outputs
```
Comment (averissimo) [inline vignettes/deepdive.Rmd:42]: CSS comments start with `/*` and end with `*/`

```suggestion
/* Used to control DT outputs */
```

File: vignettes/deepdive.Rmd [hunk 20]
```diff
@@ -0,0 +1,438 @@
...
+      list(width = "10px", targets = c("order", "length", "format", "type", "dataset", "variable")),
+      list(className = "text-left", targets = c("label"))
+    ),
+    searching = FALSE,
+    autoWidth = TRUE
+  )
+) %>%
+  formatStyle(0,
+    target = "row", color = "black", backgroundColor = "white",
+    fontWeight = "bold", lineHeight = "70%", textAlign = "center"
```
Comment (averissimo) [inline vignettes/deepdive.Rmd:232]: Bold font and line height looks weird. I suggest a middle ground on the weight between normal and bold.

```suggestion
    fontWeight = "500", lineHeight = "85%", textAlign = "center"
```

image

File: vignettes/xportr.Rmd [hunk 21]
```diff
@@ -64,191 +86,179 @@ library(dplyr)
...
-Please note that the order of the `ADSL` variables, see above, does not match specification file order column.  We can quickly remedy this with a call to `xportr_order()`.  Note that the variable `SITEID` has been moved as well as many others to match the specification file order column.
+Please note that the order of the `ADSL` variables, see above, does not match the specification file `order` column.  We can quickly remedy this with a call to `xportr_order()`.  Note that the variable `SITEID` has been moved as well as many others to match the specification file order column. Variables not in the spec are moved to the end of the data and a message is written to the console.
 
-```{r, warning=FALSE, message=FALSE, echo = TRUE, results='hide'}
+```{r, echo = TRUE}
 adsl_order <- xportr_order(adsl, var_spec, domain = "ADSL", verbose = "message")
 ```
 
 ```{r, echo = FALSE}
 DT::datatable(adsl_order, options = list(
```
Comment (averissimo) [inline vignettes/xportr.Rmd:191]: Are rownames necessary here?

```suggestion
DT::datatable(adsl_order, rownames = FALSE, options = list(
```

File: vignettes/xportr.Rmd [hunk 22]
```diff
@@ -64,191 +86,179 @@ library(dplyr)
...
   autoWidth = FALSE, scrollX = TRUE, pageLength = 5,
   lengthMenu = c(5, 10, 15, 20)
-))
+)) %>%
+  formatStyle(0, target = "row", color = "black", backgroundColor = "white", fontWeight = "bold")
```
Comment (averissimo) [inline vignettes/xportr.Rmd:195]:
```suggestion
)) %>%
  formatStyle(0, target = "row", color = "black", backgroundColor = "white", fontWeight = "500", lineHeight = "500")
```

**Comment by averissimo** [comment]
@bms63 and all,

Made a bunch of suggestions for datatables in a separate PR that merges to this one, check it out

* PR #160 
* site: https://averissimo.github.io/xportr/

File: R/type.R (hunk 16 above)
Comment (bms63) [reply R/type.R]: I reverted this back to how it was in `devel`

**Comment by bms63** [comment]
@atorus-research/xportr-development-team - whew!! this was a big one.  TY for all the reviews and code suggestions.

I think this one is good to go now.  Unless folks want to take a final peak https://bms63.github.io/xportr/index.html.

I think it will be good to see my version of the website published with the docs updates coming from https://github.com/atorus-research/xportr/pull/148 as well

File: vignettes/deepdive.Rmd (hunk 8 above)
[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/deepdive.Rmd:398]: I have updated this to _At the time of `{xportr} v0.3.0` we have not implemented any warnings or error messaging for this function. However, `xportr_write()` through `xpt_validate()` will check that formats applied are valid SAS formats._
